### ISSUE TITLE
Murmur does not seem to use parameters from ini file

### ISSUE THREAD

**Issue by hundrex** [mumble-voip/mumble#5198]
**Describe the bug**
When starting murmur, it shows in log that it is `Initializing settings from /etc/mumble-server.ini (basepath /etc)` but when starting it does not uses the port in the ini fille

**Steps to Reproduce**

1. Download murmur from repositories
2. Edit the ini file **/etc/mumble-server.ini** to change the port used (e.g. put port 66666)
3. Start murmur with the following command: `sudo /usr/sbin/murmurd -ini /etc/mumble-server.ini`
4. Consult log file (/var/log/mumble-server/mumble-server.log)

**Expected behavior**
In the logs (or when showing active ports using netstat) we should see that the server is running on the port specified in the ini file :
`Server listening on 0.0.0.0:66666` 
Else, it seems to always use the default port (**64738**) and I don't know why as I specify the ini file with -ini **and** it shows that it is using it.

**Desktop :**
 - OS: `Ubuntu 18.04.2 LTS`
 - Version: `murmurd 1.3.4-1~ppa2~bionic1`

**Additional context**
I had a functional installation of a previous version and upgraded from it using repos. I backed up my old ini file to get the new one, and copied the small things I had in it before (mainly port number and server welcome message).
I also tried to change other visible parameters (like the host so I could see if it was used with netstat) but it also does not use it.

**Comment by Krzmbrzl** [comment]
How do you specify the port inside the ini? Did you make sure that there is no other place inside the ini file that specifies the port (overwriting your definition)?

**Comment by hundrex** [comment]
In the ini, i wrote: 

```
; Port to bind TCP and UDP sockets to.
port=66666
```


I verified and there is no other line specifying the port in the ini file.

**Comment by Kissaki** [comment]
Did you reuse the old database? Did you reproduce this without a database/with a new database being created?

The ini provides default configuration. The Mumble server process is capable of running multiple virtual servers. By default a single virtual server is used. Virtual server configuration can overwrite the default provided by the ini. So if you previous virtual server configuration has a port set, changing the default in the ini does not lead to an effective change on that vserver.

**Comment by hundrex** [comment]
I indeed was reusing the old database (as I want to keep everything I created before, like users, ACLs, channels,...). Its path is the one by default, and the log also confirms that it is the one used: `ServerDB: Opened SQLite database /var/lib/mumble-server/mumble-server.sqlite`
But I am trying to use the port that was used with this old database, I never used the default port before.

I tried starting with a fresh database as suggested (I deleted the old one and let it create a new one) but the default port is still used despite the information in the ini file.

What about virtual servers though? How does that work, and what files could overwrite mine ?

**Comment by Kissaki** [comment]
The vserver configuration is stored in the sqlite database file. So it's not other configuration files that could be overwriting it.

**Comment by Kissaki** [comment]
Installing via apt may lead to existing config files staying and new original versions of it being saved alongside.

Please verify the `port` line is the only in your ini and not overwritten by another port declaration in the config file before or after.

**Comment by hundrex** [comment]
Which configuration file(s) are we talking about? I don't see any mention of any other configuration file than the .ini one, so I'm a little bit confused.

**Comment by emankced** [comment]
The used port `66666` is not a valid port. TCP and UDP use 16 bit unsigned integers as ports and those can represent numbers up to `2^16-1 = 65535`. `66666` is just too big. @hundrex Try to use a smaller port. This probably solves the problem.

**Comment by hundrex** [comment]
It was only an example, this is not the port I use. The port I use is in the range you gave.

**Comment by Krzmbrzl** [comment]
I can#t reproduce the described behavior with a version from current master. So either this is a bug in your version that has been fixed since it was released or there is indeed some problem in your specific setup.

Note that using a fresh database is not necessary for the port change to take effect.

Could you perhaps provide the entire log the server emits when starting up? Maybe that gives something useful :thinking: 
And just for the fun of it: Did you try using an ini file that **only** contains e.g. `port=54321`?

**Comment by Kissaki** [comment]
> Note that using a fresh database is not necessary for the port change to take effect.

That case was to check that a more specific vserver configuration does not lead to the default ini configuration being ignored.

**Comment by hundrex** [comment]
Here is my full log when executing the command `sudo /usr/sbin/murmurd -fg -v -ini /etc/mumble-server.ini` (I use -fg only for me to see the log easily):
```
<X>2021-07-22 00:01:42.733 SSL: OpenSSL version is 'OpenSSL 1.1.1  11 Sep 2018'
<W>2021-07-22 00:01:42.733 Initializing settings from /etc/mumble-server.ini (basepath /etc)
<W>2021-07-22 00:01:43.063 MetaParams: TLS cipher preference is "TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:AES256-SHA:AES128-SHA"
<C>2021-07-22 00:01:43.064 Successfully switched to uid 116
<W>2021-07-22 00:01:43.201 ServerDB: Opened SQLite database /var/lib/mumble-server/mumble-server.sqlite
<W>2021-07-22 00:01:43.201 ServerDB: Using SQLite's default rollback journal.
<W>2021-07-22 00:01:43.204 Resource limits were 0 0
<W>2021-07-22 00:01:43.204 Successfully dropped capabilities
<W>2021-07-22 00:01:43.440 Murmur 1.3.3 (1.3.4-1~ppa2~bionic1) running on X11: Ubuntu 18.04.2 LTS: Booting servers
<W>2021-07-22 00:01:43.483 1 => Server listening on 0.0.0.0:64738
<W>2021-07-22 00:01:43.850 1 => Announcing server via bonjour
<W>2021-07-22 00:01:43.908 1 => Not registering server as public
```

**Comment by Krzmbrzl** [comment]
I can spot nothing our of the ordinary on the logs.

Did you verify that mumble has the permission to read the INI? I would assume that thanks to sudo it should, but double-checking is probably still a good idea.

You could cross-check by copying the INI to your home directory and then starting mumble from there (without sudo).

Also did you now try to use an INI with only the port option?

**Comment by hundrex** [comment]
Okay, so. I feel a really dumb right now, because I found the source of the problem, and it was entirely my fault.

I had the good permissions, but I checked as suggested with a ini file that contained only the **port** option, and it worked, both with or without using root.
So I took the ini file I want to use, and started commenting each option one by one, restarting murmur each time to see if it would start on the custom port I gave it. And then I saw it, before I even commented the line I finally saw my mistake and knew it would then work: when I copied the text of the welcome option from the ini I previously used, I forgot to delete one of the quotes. So as the line ended with two quotes instead of one, my file was broken from there, and that was probably why murmur was then using all the default parameters, no matter what customization I made.

Sorry for wasting you time everyone, but **thanks a lot** for your reactivity and taking the time to help me, it lead me on the right tracks to find my obvious mistake.
